validators/between.py

```diff
@@ -1,61 +1,103 @@
-    not just integers.
+    not just integers. This validator is originally based on `WTForms-NumberRange-Validator`_
 
-    This validator is originally based on `WTForms NumberRange validator`_.
-
-    .. _WTForms NumberRange validator:
-       https://github.com/wtforms/wtforms/blob/master/wtforms/validators.py
+    .. _WTForms-NumberRange-Validator:
+        https://github.com/wtforms/wtforms/blob/master/src/wtforms/validators.py#L166-L220
 
     Examples::
 
         >>> from datetime import datetime
 
-        >>> between(5, min=2)
-        True
+        >>> between(5, min_val=2)
+        # Output: True
 
-        >>> between(13.2, min=13, max=14)
-        True
+        >>> between(13.2, min_val=13, max_val=14)
+        # Output: True
 
-        >>> between(500, max=400)
-        ValidationFailure(func=between, args=...)
+        >>> between(500, max_val=400)
+        # Output: ValidationFailure(func=between, args=...)
 
         >>> between(
         ...     datetime(2000, 11, 11),
-        ...     min=datetime(1999, 11, 11)
+        ...     min_val=datetime(1999, 11, 11)
         ... )
-        True
+        # True
+
+    Args:
+        `value`:
+            [Required] Value which is to be compared.
+        `min_val`:
+            [Optional] The minimum required value of the number.
+            If not provided, minimum value will not be checked.
+        `max_val`:
+            [Optional] The maximum value of the number.
+            If not provided, maximum value will not be checked.
+        Either one of `min_val` or `max_val` must be provided.
+
+    Returns:
+        A `boolean` if `value` is greater than `min_val` and
+        less than `max_val`.
 
-    :param min:
-        The minimum required value of the number. If not provided, minimum
-        value will not be checked.
-    :param max:
-        The maximum value of the number. If not provided, maximum value
-        will not be checked.
+    Raises:
+        `AssertionError`:
+        - If both `min_val` and `max_val` are `None`.
+        - If `min_val` is greater than `max_val`.
+
+        `TypeError`:
+        - If there's a type mismatch before comparison
 
     .. versionadded:: 0.2
     """
-    if min is None and max is None:
-        raise AssertionError(
-            'At least one of `min` or `max` must be specified.'
-        )
-    if min is None:
-        min = Min
-    if max is None:
-        max = Max
-    try:
-        min_gt_max = min > max
-    except TypeError:
-        min_gt_max = max < min
-    if min_gt_max:
-        raise AssertionError('`min` cannot be more than `max`.')
-
-    return min <= value and max >= value
+    if min_val is None and max_val is None:
+        raise AssertionError("At least one of either `min_val` or `max_val` must be specified")
+
+    if max_val is None:
+        max_val = AbsMax()
+    if min_val is None:
+        min_val = AbsMin()
+
+    # if isinstance(min_val, AbsMin) and isinstance(max_val, AbsMax):
```

Comment:
Why is this code block commented out? Should it be removed?

Reply:
Hi @kvesteri, as you know, that the commented condition in line 83 is `True`, only if both, `min_val` and `max_val` are `None`. But that will not be true since we're raising an `AssertionError` on line 76.

```py
if min_val is None and max_val is None:
        raise AssertionError("At least one of either `min_val` or `max_val` must be specified")
```

I was thinking, later on if we 'decide' to remove the `AssertionError`, so that `validators.between(10)` returns either `True` or `False`, we might needing that condition. And then you may ask: _'why change the function's behavior?'_

I don't know yet, I'll have to look into the other modules to find a consistency or pattern of behavior.

Reply:
I think if you are unsure it is better to make pull request for review and ask for comments rather than merging straight to master. Now we have code in master which is work-in-progress rather than stable.

Reply:
The comment is just for future reference. Both modules `between` and `length` are **_not_** 'work-in-progress', they are 'stable'.

> I think if you are unsure it is better to make pull request for review and ask for comments

Certainly, if I'm unsure about the changes, I will only _draft_ PRs and ask for help.

---

Further, I keep a PR open-for-review for about 24 hours after creation, should I increase the time?

Also, I'm not aware of your workflow regarding git branches. For me I keep things simple.

- `master` is same as release branch. Stable releases are marked by tags.
- `workshop` is same as develop branch.

If you've some specific workflow, you'd prefer for this repository please let me know.

Reply:
So in general I don't like commented out code. It feels like WIP. So I think that when there is a case where you need to comment out lines we should make a decision about those lines and not commit to master TODO or WIP lines.